libethashseal/Ethash.cpp

```diff
@@ -247,3 +247,76 @@ bool Ethash::shouldSeal(Interface*)
 {
     return true;
 }
+
+bool Ethash::isMining() const
+{
+    return m_farm.isMining();
+}
+
+WorkingProgress Ethash::miningProgress() const
+{
+    if (isMining())
+        return m_farm.miningProgress();
+    return WorkingProgress();
+}
+
+u256 Ethash::hashrate() const
+{
+    u256 r = externalHashrate();
+    if (isMining())
+        r += miningProgress().rate();
+    return r;
+}
+
+std::tuple<h256, h256, h256> Ethash::getEthashWork(Client const& _client)
+{
+    // lock the work so a later submission isn't invalidated by processing a transaction elsewhere.
+    // this will be reset as soon as a new block arrives, allowing more transactions to be
+    // processed.
+    bool oldShould = shouldServeWork(_client);
+    m_lastGetWork = chrono::system_clock::now();
```

Comment:
1. Add `getWork` to `SealEngineFace`, rename `getEthashWork` to `getWork` (and you'll need some dummy implementation in `NoProof`)
2. Then create `Client::getWork` that will update `m_lastGetWork` like here and then call `sealEngine()->getWork()`
3. Call `Client::getWork` from `eth_getWork`

Reply:
1. -> 
sealEngineFace could not include Client. I could not declare 
```std::tuple<h256, h256, h256> SealEngineFace::getWork(Client const& _client)```